### PR TITLE
SDK-2055-2072: Add Identity Profile Requirements to DynamicPolicyBuil…

### PR DESCRIPTION
**Pull request by BenSmithYoti**
…der. Add Subject to DynamicScenarioBuilder. Add relevant tests. Modify SignedMessageFactory to remove unncessary (& warning producing) ToString.

### CODE REVIEW

File: src/Yoti.Auth/Web/SignedMessageFactory.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public static string SignMessage(HttpMethod httpMethod, string endpoint, Asymmet
                 throw new ArgumentNullException(nameof(httpMethod));
             }
 
-            string stringToConvert = $"{httpMethod.ToString()}&{endpoint}";
+            string stringToConvert = $"{httpMethod}&{endpoint}";
```
Comment (echarrod) [inline src/Yoti.Auth/Web/SignedMessageFactory.cs:16]: Presume this was an unnecessary cast?

File: test/Yoti.Auth.Tests/ShareUrl/Policy/DynamicPolicyBuilderTests.cs [hunk 2]
```diff
@@ -312,5 +312,18 @@ public void ShouldBuildWithRememberMeFlag()
 
             Assert.IsTrue(result.WantedRememberMeId);
         }
+
+
```
Comment (echarrod) [inline test/Yoti.Auth.Tests/ShareUrl/Policy/DynamicPolicyBuilderTests.cs:316]:
```suggestion
```

File: src/Yoti.Auth/Web/SignedMessageFactory.cs (hunk 1 above)
Comment (BenSmithYoti) [reply src/Yoti.Auth/Web/SignedMessageFactory.cs:16]: yes indeed totally unecessary ToString() call inside interpolation, plus was kicking out warning.

**Review by laurent-yoti — COMMENTED** [review]
Makes sense to me, can not really comment on code style though.

**Review by echarrod — COMMENTED** [review]
Looks good. Let's wait until we can test before merging though :+1:

**Review by echarrod — APPROVED** [review]
Thinking about it, will be a pain to test if we are doing all separate PRs. Happy for this to be merged and then further iterations made if needed

**Pull request merged**